From ede68b56509109b1f391214a15b9ed73791e7d3c Mon Sep 17 00:00:00 2001
From: BirdZhang <0312birdzhang@gmail.com>
Date: Fri, 6 Dec 2024 21:51:21 +0800
Subject: [PATCH] Install files from /vendor

---
 droid-hal-device.inc | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/droid-hal-device.inc b/droid-hal-device.inc
index 68228cd..3ef0d31 100644
--- a/droid-hal-device.inc
+++ b/droid-hal-device.inc
@@ -661,6 +661,7 @@ sed -i -e '/^[[:space:]]*mkdir[[:space:]]\/tmp[[:space:]]*/s/^/# Removed during
 
 if [ $android_version_major -ge "7" ]; then
 mkdir -p $RPM_BUILD_ROOT%{_libexecdir}/droid-hybris/system/etc/init
+mkdir -p $RPM_BUILD_ROOT%{_libexecdir}/droid-hybris/vendor/
 cp -a %{android_root}/out/target/product/%{device}/system/etc/init/servicemanager.rc $RPM_BUILD_ROOT%{_libexecdir}/droid-hybris/system/etc/init/
 echo "%{_libexecdir}/droid-hybris/system/etc/init/*.rc" > tmp/droid-hal.files
 
@@ -693,8 +694,10 @@ echo "/default.prop" >> tmp/droid-hal.files
 fi
 
 cp -a %{android_root}/out/target/product/%{device}/system/{bin,lib} $RPM_BUILD_ROOT%{_libexecdir}/droid-hybris/system/.
+cp -a %{android_root}/out/target/product/%{device}/vendor/lib $RPM_BUILD_ROOT%{_libexecdir}/droid-hybris/vendor/.
 %if 0%{?droid_target_aarch64:1}
-cp -a %{android_root}/out/target/product/%{device}/system/lib64 $RPM_BUILD_ROOT%{_libexecdir}/droid-hybris/system/.
+cp -ar %{android_root}/out/target/product/%{device}/system/lib64 $RPM_BUILD_ROOT%{_libexecdir}/droid-hybris/system/.
+cp -ar %{android_root}/out/target/product/%{device}/vendor/lib64 $RPM_BUILD_ROOT%{_libexecdir}/droid-hybris/vendor/.
 %endif
 if [ $android_version_major -ge "10" ]; then
     # Create empty /apex, /system/bin/apexd will then mount namespaces therein
@@ -1144,11 +1147,13 @@ done
 %{_unitdir}
 # hybris and /dev/alog/ libraries
 %{_libexecdir}/droid-hybris/system/lib/
+%{_libexecdir}/droid-hybris/vendor/lib/
 # just /dev/alog/ libraries (for trying to run pure android apps)
 %{_libexecdir}/droid-hybris/lib-dev-alog/.
 # all of the above for 64bits
 %if 0%{?droid_target_aarch64:1}
 %{_libexecdir}/droid-hybris/system/lib64/
+%{_libexecdir}/droid-hybris/vendor/lib64/
 %{_libexecdir}/droid-hybris/lib64-dev-alog/.
 %endif
 /lib/udev/rules.d/*
-- 
2.25.1

